tests/python/test_json_schema_converter.py

```diff
@@ -2192,7 +2193,7 @@ def test_limited_whitespace_cnt():
 """
     schema = {"type": "object", "properties": {"key": {"type": "string"}}, "required": ["key"]}
     grammar = xgr.Grammar.from_json_schema(schema, any_whitespace=True, max_whitespace_cnt=2)
-
+    grammar = GrammarFunctor.grammar_optimizer(grammar)
```

Comment:
For this one, we can keep this instead of updating the expected grammar because we want to expect the grammar end-to-end